lib/route_translator/translator/path/segment.rb

```diff
@@ -24,16 +27,24 @@ def translate_string(str, locale)
         # "people(.:format)", only "people" will be translated.
         # If there is no translation, the path segment is blank, begins with a
         # ":" (param key) or "*" (wildcard), the segment is returned untouched.
-        def translate(segment, locale)
+        def translate(segment, locale, scope)
           return segment if segment.empty?
           named_param, hyphenized = segment.split('-'.freeze, 2) if segment.starts_with?(':'.freeze)
-          return "#{named_param}-#{translate(hyphenized.dup, locale)}" if hyphenized
+          return "#{named_param}-#{translate(hyphenized.dup, locale, scope)}" if hyphenized
           return segment if segment.starts_with?('('.freeze) || segment.starts_with?('*'.freeze) || segment.include?(':'.freeze)
 
           appended_part = segment.slice!(/(\()$/)
           match = TRANSLATABLE_SEGMENT.match(segment)[1] if TRANSLATABLE_SEGMENT.match(segment)
 
-          (translate_string(match, locale) || segment) + appended_part.to_s
+          (translate_string(match, locale, scope) || segment) + appended_part.to_s
+        end
+
+        def fallback_options(str, locale)
```

Comment:
fallback_options is not used outside this module, so you should move it inside `class << self`